x-pack/plugins/security_solution/public/common/components/exceptions/builder/index.tsx

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useMemo, useCallback, useEffect, useState } from 'react';
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import styled from 'styled-components';
+
+import { ExceptionListItemComponent } from './exception_item';
+import { useFetchIndexPatterns } from '../../../../alerts/containers/detection_engine/rules/fetch_index_patterns';
+import {
+  ExceptionListItemSchema,
+  NamespaceType,
+  exceptionListItemSchema,
+  OperatorTypeEnum,
+  OperatorEnum,
+  CreateExceptionListItemSchema,
+} from '../../../../../public/lists_plugin_deps';
+import { AndOrBadge } from '../../and_or_badge';
+import { BuilderButtonOptions } from './builder_button_options';
+import { getNewExceptionItem } from '../helpers';
+import { ExceptionsBuilderExceptionItem, CreateExceptionListItemBuilderSchema } from '../types';
+import { Loader } from '../../loader';
+
+const MyInvisibleAndBadge = styled(EuiFlexItem)`
+  visibility: hidden;
+`;
+
+const MyAndBadge = styled(AndOrBadge)`
+  & > .euiFlexItem {
+    margin: 0;
+  }
+`;
+
+const MyButtonsContainer = styled(EuiFlexItem)`
+  margin: 16px 0;
+`;
+
+interface OnChangeProps {
+  exceptionItems: Array<ExceptionListItemSchema | CreateExceptionListItemSchema>;
+  exceptionsToDelete: ExceptionListItemSchema[];
+}
+
+interface ExceptionBuilderProps {
+  exceptionListItems: ExceptionListItemSchema[];
+  listType: 'detection' | 'endpoint';
+  listId: string;
+  listNamespaceType: NamespaceType;
+  ruleName: string;
+  indexPatternConfig: string[];
+  isLoading: boolean;
+  isOrDisabled: boolean;
+  isAndDisabled: boolean;
+  onChange: (arg: OnChangeProps) => void;
+}
+
+export const ExceptionBuilder = ({
+  exceptionListItems,
+  listType,
+  listId,
+  listNamespaceType,
+  ruleName,
+  indexPatternConfig,
+  isLoading,
+  isOrDisabled,
+  isAndDisabled,
+  onChange,
+}: ExceptionBuilderProps) => {
+  const [andLogicIncluded, setAndLogicIncluded] = useState<boolean>(false);
```

Comment:
This is used to determine when items need to be indented appropriately.

If no and logic:
![Screen Shot 2020-06-30 at 9 44 08 PM](https://user-images.githubusercontent.com/10927944/86193793-d8db0e80-bb1a-11ea-8223-8a61eed976b2.png)

If and logic:
![Screen Shot 2020-06-30 at 9 40 47 PM](https://user-images.githubusercontent.com/10927944/86193800-e09ab300-bb1a-11ea-96cc-461ca302c009.png)